src/library/searchquery.cpp

```diff
@@ -107,6 +107,10 @@ NotNode::NotNode(QueryNode* pNode)
         : m_pNode(pNode) {
 }
 
+NotNode::~NotNode() {
+    delete m_pNode;
```

Comment:
This question becomes obsolete with the refactoring for master which consequently uses std::unique_ptr. Another nice side effect of C++11 :)